### ISSUE TITLE
Support for azurerm_storage_account not to access data-plane

### ISSUE THREAD

**Issue by mvervoort** [hashicorp/terraform-provider-azurerm#16727]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

**Background**
When deploying a Storage Account from an agent that is behind a corporate firewall that is prohibited access to `*.blob.core.windows.net` and `*.queue.core.windows.net`, the deployment will fail. 

After taking 5 minutes trying to connect to mystorage.queue.core.windows.net the deployment fails with error: 
`Error: reading queue properties for AzureRM Storage Account "mystorage": queues.Client#GetServiceProperties: Failure sending request: StatusCode=0 -- Original Error: context deadline exceeded`

This is initiated by #16240, and related to #2977 and #9314, but different as it's about a corporate firewall. When this firewall is temporarily allowed access to `*.blob.core.windows.net` and `*.queue.core.windows.net`, it can continue to deploy the Storage Account, next it can deploy a Private Endpoint, and after a sleep of 2 minutes (yes, ugly hack) it can create containers. 

But as many large corporate enterprises, the security team has established it's a data-exfiltration risk to allow access to `*.blob.core.windows.net` and `*.queue.core.windows.net`, and therefor the corporate firewall will block access. This request is from a large Dutch enterprise building semiconductor machines.

**Request**
Add an option to not-access the data-plane (to fetch the queue properties). Perhaps because of this the functionality of azurerm_storage_account will be slightly limited, but at least it's possible to deploy and update a Storage Account using the control-plane from behind a corporate firewall.

Perhaps this can be done by specifying an additional property on the azurerm_storage_account resource like `do_not_access_data_plane`, or by setting a flag in the `azurerm` provider features section.

Optional we could try to make this change ourselves, but would love to get advice upfront. We're not skilled enough yet, but we can give it a try.

### New or Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Potential Terraform Configuration

```hcl
resource "azurerm_storage_account" "this" {
  name                     = "mystorage"
  resource_group_name      = "my-resource-group"
  location                 = "westeurope"
  do_not_access_data_plane = true
}

or

provider "azurerm" {
  features {
    storage_account {
      do_not_access_data_plane = true
    }
  }
}
```

### References

#2977
#9314
#16240

**Comment by tombuildsstuff** [comment]
Duplicate of #10765

**Comment by mvervoort** [comment]
@tombuildsstuff, thanks for the reply.
The duplicate (#10765) is regarding provisioning Containers, File Shares, Queues etc, while this request is merely for the initial storage account. After the storage account is deployed, we could deploy a Private Endpoint, and next provisioning Containers, File Shares, Queues etc.

Does it make sense to adjust the azurerm_storage_account module (by ourselves) by including an optional flag like do_not_access_data_plane (or skip_data_plane), to make sure a storage account can be deployed using Terraform with a build-agent behind a corporate firewall (and thereby reducing the functionality of the azurerm_storage_account slightly)?

**Comment by marco-vervoort-asml** [comment]
Any update on this?

**Comment by tombuildsstuff** [comment]
@mvervoort @marco-vervoort-asml as above this is blocked on the Service Team since the API's aren't compatible - as such we have to reach out to the Data Plane API's here, meaning that (as above) we can't add a feature-flag for this.

In this case your best bet is going to be reaching out to the Service Team, once these two API's are compatible we should be able to ship this pretty quickly - but for now unfortunately this is blocked on the Service Team.

**Comment by SovakPaleny** [comment]
@tombuildsstuff can you please elaborate on what needs to be read directly from the queues data plane as part of the `azurerm_storage_account` resource?
